sqlglot/dialects/dialect.py

```diff
@@ -42,6 +42,20 @@ class Dialects(str, Enum):
 class _Dialect(type):
     classes: t.Dict[str, t.Type[Dialect]] = {}
 
+    def __eq__(cls, other: DialectType) -> bool:  # type: ignore
```

Comment:
why is it type ignore? other should be t.Any not DialecTType

Reply:
fixed